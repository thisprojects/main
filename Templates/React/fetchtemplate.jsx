import React from 'react';
import Fetch from 'react-fetch';






class Welcome extends React.Component {
    render (){
        return (<h1>Hello1 {this.props.value}</h1>);


    }

}

class List extends React.Component {

                render () {
   return (
       <ul>
           {this.props.list.map(station => <li key={station}> {station} </li>)}
       </ul>
   );
}



}


class ShoppingList extends React.Component {

constructor (props){
            super(props);
            this.state = {value: "hello", schnups: false, list: ["one","two","three"], results: [] }

        }

componentDidMount () {

 this._fetchIT ();

}


  render() {



    return (
      <div className="shopping-list">
        <h1>List of Stuff</h1>

        <ul>
        <List list={this.state.list}/>
        </ul>
        <Welcome value={this.state.value}/>
        {this._doStuff()}
        {this._jsonList()}
        <div>

        </div>
        <button onClick={this._testSt.bind(this, this.state.schnups)}>Click me!</button>
      </div>

    );
  }

_doStuff () {

    if (this.state.schnups){

    return (<h1> Bibobs </h1>);
     }
}

_testSt () {
        this.setState({value: "wibbums", schnups: !this.state.schnups, list: ["one", "two", "three", "four", 13] });
        this.setState({results: []});
        this._fetchIT();

    }
_jsonList (){

    if (this.state.results.length){
        let starwars = this.state.results
        return (
            <div>

                {starwars.map (starwars  =>
                <ul>
                <li key={starwars.toString()}>{starwars.name}</li>
                </ul>
                )}
            </div>
        )


    }else{

        return (

            <div>
            <h1>Loading!...</h1>
            </div>
        )
    }


}

_fetchIT () {
    fetch('https://swapi.co/api/people/')
   .then(response => response.json())
      .then(data => this.setState({ results: data.results }));


}

}

export default ShoppingList;
