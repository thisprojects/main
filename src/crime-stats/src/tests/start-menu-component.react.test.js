import React from 'react';
import { 
  PostCode,
  Slider,
  SubmitButton,
  Year,
  YearPicker,
  Month,
  MonthPicker 
} from '../main-components/start-menu-components.js';

import { faPlayCircle } from '@fortawesome/free-solid-svg-icons';
import renderer from 'react-test-renderer';


test('PostCode Matches Snapshot', () => {
  const component = renderer.create(
    <PostCode />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Slider Matches Snapshot', () => {
  const component = renderer.create(
    <Slider showAndHideSliders={"stub"} selector={"#header-wrapper"} theClass={"header-stub"} label={"Start"} slideClass={"side-hide"} icon={faPlayCircle}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('SubmitButton Matches Snapshot', () => {
  const component = renderer.create(
    <SubmitButton />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Year Matches Snapshot', () => {
  const component = renderer.create(
    <Year />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('YearPicker Matches Snapshot', () => {
  const component = renderer.create(
    <YearPicker years={["1","2","3","4","5"]} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Month Matches Snapshot', () => {
  const component = renderer.create(
    <Month />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('MonthPicker Matches Snapshot', () => {
  const component = renderer.create(
    <MonthPicker />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

