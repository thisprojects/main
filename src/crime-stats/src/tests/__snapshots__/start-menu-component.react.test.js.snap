// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month Matches Snapshot 1`] = `
<div
  className="date-month"
>
  <span>
    Month
  </span>
  <select
    id="month-select"
  >
    <option
      value="01"
    >
      01
    </option>
    <option
      value="02"
    >
      02
    </option>
    <option
      value="03"
    >
      03
    </option>
    <option
      value="04"
    >
      04
    </option>
    <option
      value="05"
    >
      05
    </option>
    <option
      value="06"
    >
      06
    </option>
    <option
      value="07"
    >
      07
    </option>
    <option
      value="08"
    >
      08
    </option>
    <option
      value="09"
    >
      09
    </option>
    <option
      value="10"
    >
      10
    </option>
    <option
      value="11"
    >
      11
    </option>
    <option
      value="12"
    >
      12
    </option>
  </select>
</div>
`;

exports[`MonthPicker Matches Snapshot 1`] = `
Array [
  <option
    value="01"
  >
    01
  </option>,
  <option
    value="02"
  >
    02
  </option>,
  <option
    value="03"
  >
    03
  </option>,
  <option
    value="04"
  >
    04
  </option>,
  <option
    value="05"
  >
    05
  </option>,
  <option
    value="06"
  >
    06
  </option>,
  <option
    value="07"
  >
    07
  </option>,
  <option
    value="08"
  >
    08
  </option>,
  <option
    value="09"
  >
    09
  </option>,
  <option
    value="10"
  >
    10
  </option>,
  <option
    value="11"
  >
    11
  </option>,
  <option
    value="12"
  >
    12
  </option>,
]
`;

exports[`PostCode Matches Snapshot 1`] = `
<div
  className="postcode-picker"
>
  <input
    id="post-code"
    placeholder="Location"
    type="text"
  />
</div>
`;

exports[`Slider Matches Snapshot 1`] = `
<div
  className="header-stub"
  onClick={[Function]}
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-play-circle fa-w-16 "
    data-icon="play-circle"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm115.7 272l-176 101c-15.8 8.8-35.7-2.5-35.7-21V152c0-18.4 19.8-29.8 35.7-21l176 107c16.4 9.2 16.4 32.9 0 42z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <p>
    Start
  </p>
</div>
`;

exports[`SubmitButton Matches Snapshot 1`] = `
<div
  className="submit-button"
>
  <button
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;

exports[`Year Matches Snapshot 1`] = `
<div
  className="date-year"
>
  <span>
    Year
  </span>
  <select
    defaultValue="2019"
    id="year-select"
  >
    <option
      value={2020}
    >
      2020
    </option>
    <option
      value={2019}
    >
      2019
    </option>
    <option
      value={2018}
    >
      2018
    </option>
    <option
      value={2017}
    >
      2017
    </option>
    <option
      value={2016}
    >
      2016
    </option>
    <option
      value={2015}
    >
      2015
    </option>
    <option
      value={2014}
    >
      2014
    </option>
    <option
      value={2013}
    >
      2013
    </option>
    <option
      value={2012}
    >
      2012
    </option>
    <option
      value={2011}
    >
      2011
    </option>
    <option
      value={2010}
    >
      2010
    </option>
  </select>
</div>
`;

exports[`YearPicker Matches Snapshot 1`] = `
Array [
  <option
    value="1"
  >
    1
  </option>,
  <option
    value="2"
  >
    2
  </option>,
  <option
    value="3"
  >
    3
  </option>,
  <option
    value="4"
  >
    4
  </option>,
  <option
    value="5"
  >
    5
  </option>,
]
`;
